core/src/env.rs

```diff
@@ -192,17 +186,16 @@ impl<E: Ext> LaterExt<E> {
     }
 
     /// Unset inner ext
-    pub fn unset(&mut self) -> E {
-        self.inner
-            .borrow_mut()
-            .take()
-            .expect("Unset should be paired with set and called after")
+    pub fn take(self) -> Option<E> {
```

Comment:
I'd rename this to `into_inner` because `take` usually implies reference

Reply:
will do in #853, ok? just want to merge it asap + I checked out the new PR from this branch.